### PR TITLE
Custom deployment updates

### PR DESCRIPTION
**Pull request by jasonmj**
Hello friends! I've recently worked through a custom Nerves Hub deployment for the fine folks at [ROAR for Good](https://github.com/ROARforGood/). They we're kind enough to give me permission to share the things I learned along the way for others in the community to benefit from. 

In this PR, I've added a few key changes to the `terraform.tfvars.example` and `app/vpc.tf` files regarding specific version declarations required. I've also included a copy of the `nix-shell` configuration file I used to create a shell environment with the correct dependencies for running the code. Feel free to discard that if it doesn't fit with your vision for the project. I will note that the `terraform` version, in particular, was very important to set correctly. With newer versions of `terraform` and/or `awscli` I was getting all kinds of deprecation warnings. 

One more important note here is that that commented line for `device_image` tripped me up since it's got the wrong image name. I've corrected that here in hopes that it will save others time and headache if they choose to uncomment those lines during a custom deployment.

### CODE REVIEW

File: app/vpc.tf
Comment (dnrce) [inline app/vpc.tf]: Just style, but isn't this `~> 2.4`?

File: terraform.tfvars.example
Comment (dnrce) [inline terraform.tfvars.example]: Given the Postgres versioning policy and the use of `auto_minor_version_upgrade = true`, can we specify only the major version here? Otherwise Terraform is going to try to roll things back if applied again after a minor-version auto-upgrade.

Also, does it make the most sense for this variable to be overridden here by default? The variable is given a default value `11.4` when defined. If this needs to be 12, should that default be updated instead?

File: shell.nix
Comment (dnrce) [inline shell.nix:1]: A number of other NervesHub and Nerves projects use asdf and `.tool-versions`. I'll let the maintainers weigh in but I think following that standard might be preferable to adding nix as a one-off.

**Review by dnrce — COMMENTED** [review]
Thanks for sharing your learnings!

File: app/vpc.tf
Comment (jasonmj) [reply app/vpc.tf]: I'm pretty sure we need to enforce the second half (`< 3.0.0`) as well, so that nothing newer gets used (since we're operating withing a finite window of compatible versions). I could certainly be wrong.

File: terraform.tfvars.example
Comment (jasonmj) [reply terraform.tfvars.example]: 12.10 was the earliest working version I could find. The default of 11.4 caused an error where AWS complained the db version was not available for the instance class. It seems their offerings change from time to time and this variable and/or it's default may continue to need to be updated.

Comment (dnrce) [reply terraform.tfvars.example]: I suspect it was the minor version of Postgres 11, not the use of 11, per se. It is still possible to provision 11.12 (and 9.6.22, for that matter). Or were you using an ARM instance class, which requires Postgres >=12.3?

Either way, I think this is a sensible update, but a) should better be updated as a default in `setup/variables.tf` and b) should be major-version only.

File: app/vpc.tf
Comment (dnrce) [reply app/vpc.tf]: https://www.terraform.io/language/expressions/version-constraints#-3

Comment (jasonmj) [reply app/vpc.tf]: I've just pushed an update with the change to `~2.4`.

File: terraform.tfvars.example
Comment (jasonmj) [reply terraform.tfvars.example]: I've just pushed an update that reverts my moves my changes from `terraform.tfvars.example` to `setup/variables.tf` with a major version only.

**Review by jjcarstens — APPROVED** [review]
Omg im so sorry @jasonmj ! I completely missed this. Thanks for the help 💗

**Comment by jasonmj** [comment]
No worries. Heads up: I may have another PR for you in the near future.

**Pull request merged**
